services/lambda_/drivers/LambdaCommon.py

```diff
@@ -230,4 +231,25 @@ def _check_function_in_used(self):
                 self.results['lambdaNotInUsed' + str(day) + 'Days'] = [-1, '']
                 return
 
+        return
+    
+    def _check_function_public_access(self):
+        try:
+            results = self.lambda_client.get_policy(
+                FunctionName=self.function_name
+            )
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                return
+            else:
+                raise e
+                
+        if results.get('Policy'):
+            doc = json.loads(results.get('Policy'))
+            pObj = Policy(doc)
+            pObj.inspectPrinciple()
+            
+            if pObj.hasPublicAccess() == True:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)